lib/reek/smells/smell_detector.rb

```diff
@@ -38,10 +38,19 @@ def contexts
       end
 
       def run_for(context)
-        return [] unless enabled_for?(context)
         return [] if exception?(context)
 
-        sniff(context)
+        smells = sniff(context)
+
+        if enabled_for?(context)
```

Comment:
You're checking for a boolean but have 3 branches which is confusing.
I'd prefer

```Ruby
if
else
  if
  else
  end
end
```